internal/services/appconfiguration/app_configuration_key_resource_test.go

```diff
@@ -0,0 +1,85 @@
+package appconfiguration_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/appconfiguration/parse"
+
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+)
+
+type AppConfigurationKeyResource struct {
+}
+
+func TestAccAppConfigurationKey_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_app_configuration_key", "test")
+	r := AppConfigurationKeyResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
```

Comment:
can we add a Requires Import & an update test here (in particular when there's a Lock involved, adding/removing that lock)